### PR TITLE
Improve pg_SUITE

### PR DESCRIPTION
**Pull request by sverker**
Fix sometimes failing test cases.

@max-au

### CODE REVIEW

File: lib/kernel/test/pg_SUITE.erl [hunk 1]
```diff
@@ -715,7 +726,18 @@ sync(GS) ->
...
 sync_via({RegName, Node}, GS) ->
-    rpc:call(Node, sys, replace_state, [RegName, fun (S) -> (catch sys:get_state(GS)), S end]).
+    MyNode = node(),
+    rpc:call(Node, sys, replace_state,
+             [RegName, fun (S) -> (catch sys:get_state({GS,MyNode})), S end]);
+
+%% flush remote GS queue from local process RegName
+sync_via(RegName, {GS, Node}) ->
+    sys:replace_state(RegName,
+                      fun (S) -> _R = (catch sys:get_state({GS,Node})),
```
Comment (max-au) [inline lib/kernel/test/pg_SUITE.erl:736]: Same formatting nit, `{GS, Node}`

**Review by max-au — APPROVED** [review]
Thank you for fixing these issues!

File: lib/kernel/test/pg_SUITE.erl [hunk 2]
```diff
@@ -67,13 +67,46 @@ suite() ->
...
+            Mon = erlang:monitor(process, Tracer),
+            Tracer ! flush,
+            normal = receive
+                         {'DOWN', Mon, process, Tracer, R} -> R
+                     end
+    end.
+
+tracer() ->
+    receive flush -> ok end,
+    io:format("Flush trace messages:\n"),
```
Comment (max-au) [inline lib/kernel/test/pg_SUITE.erl:98]: Do you mind removing these debug messages in the final version?

Comment (sverker) [reply lib/kernel/test/pg_SUITE.erl:98]: Yes, that was the plan. Just want to make sure pg_SUITE runs clean for a while before I merge.
Thanks for the reminder.

Comment (sverker) [reply lib/kernel/test/pg_SUITE.erl:98]: I might keep the trace code deactivated, if that's ok with you. I have already rewritten it twice.

Comment (max-au) [reply lib/kernel/test/pg_SUITE.erl:98]: That'd be fine, but then could you move it next to `sync`/`sync_via` functions to preserve the overall structure of the suite?

**Pull request merged**